Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Booking.hs

```diff
@@ -107,9 +117,16 @@ findFareForCancelledBookings :: MonadFlow m => [Id Booking] -> m Money
 findFareForCancelledBookings bookingIds = findAllWithKV [Se.And [Se.Is BeamB.status $ Se.Eq CANCELLED, Se.Is BeamB.id $ Se.In $ getId <$> bookingIds]] <&> sum . map Domain.Types.Booking.estimatedFare
 
 instance FromTType' BeamB.Booking Booking where
+  fromTType' :: MonadFlow m => BeamB.Booking -> m (Maybe Booking)
   fromTType' BeamB.BookingT {..} = do
-    fl <- QBBL.findById (Id fromLocationId) >>= fromMaybeM (InternalError $ "FromLocation not found in booking for fromLocationId: " <> show fromLocationId)
-    tl <- QBBL.findById (Id toLocationId) >>= fromMaybeM (InternalError $ "ToLocation not found in booking for toLocationId: " <> show toLocationId)
+    mappings <- QLM.findByEntityId id
+    let fromLocationMapping = filter (\loc -> loc.order == 0) mappings
+        toLocationMappings = filter (\loc -> loc.order /= 0) mappings
+    when (null fromLocationMapping || null toLocationMappings) $ throwError (InternalError "Entity Mappings Not Found")
+    let fromLoc = head fromLocationMapping
```

Comment:
you can avoid partial `head` function:

`listToMaybe fromLocationMapping & fromMaybeM  (InternalError "Entity to location Mappings Not Found")`

or even 
```
case fromLocationMapping of
  [] -> throwError (InternalError "Entity from location mappings not found")
  [x] -> pure x
  _ ->  throwError (InternalError "Multiple from location Mappings)
```

Also proper error codes can be added